frontend/public/lib/socialMediaOperations.js

```diff
@@ -0,0 +1,138 @@
+import InstagramIcon from "@material-ui/icons/Instagram";
+import TwitterIcon from "@material-ui/icons/Twitter";
+import YouTubeIcon from "@material-ui/icons/YouTube";
+import FacebookIcon from "@material-ui/icons/Facebook";
+import LinkedInIcon from "@material-ui/icons/LinkedIn";
+
+import { apiRequest } from "./apiOperations";
+
+export function verifySocialMediaLink(socialChannel, url, texts) {
+  let error = "";
+  // matches http://, https:// , https://www. ,http://www.
+  const regexPrefix = "^(http)(?:s)?(://)(?:www.)?";
+  // matches.com/ anything
+  const regexSuffix = ".com/.+$";
+  let regex;
+  let matches = false;
+
+  switch (socialChannel.name) {
+    case "Twitter": // twitter
+      regex = new RegExp(regexPrefix + "twitter" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_twitter;
+      break;
+    case "Youtube": // youtube
+      regex = new RegExp(regexPrefix + "youtube" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_youtube;
+      break;
+    case "LinkedIn": // linkedIn
+      regex = new RegExp(regexPrefix + "linkedin" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_linkedin;
+      break;
+    case "Instagram": // instagram
+      regex = new RegExp(regexPrefix + "instagram" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_instagram;
+      break;
+    case "Facebook": // facebook
+      regex = new RegExp(regexPrefix + "facebook" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_facebook;
+      break;
+    default:
+      break;
+  }
```

Comment:
1. We should use an object and a generic function rather than switch.case here. Switch case is not meant for handling a bunch of cases that are the same except one string.
2. All of these should pass and be sent to the backend as https://twitter.com/test: "twitter.com/test", "www.twitter.com/test". The user should not be required to type out a prefix.